### PR TITLE
[ADD] Add stock_reminder

### PR DESCRIPTION
**Pull request by daramousk**

### CODE REVIEW

**Comment by rousseldenis** [comment]
@daramousk I've restarted travis job as a fix has been merged.

**Comment by rousseldenis** [comment]
@daramousk Could you rebase?

File: stock_reminder/models/stock_reminder.py [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+
+
+class StockReminder(models.Model):
+    _name = 'stock.reminder'
+
+    res_user_id = fields.Many2one(
+        'res.users',
+        string='User',
+        help='User that will be notified',
+        required=True
```
Comment (rousseldenis) [inline stock_reminder/models/stock_reminder.py:16]: ondelete='cascade' ?

File: stock_reminder/models/stock_reminder.py [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+    )
+    product_product_id = fields.Many2one(
+        'product.product',
+        string='Product',
+        help='Product to be notified for.',
+        required=True,
```
Comment (rousseldenis) [inline stock_reminder/models/stock_reminder.py:22]: ondelete='cascade' ?

File: stock_reminder/models/stock_reminder.py [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+    )
+    quantity = fields.Float(
+        required=True,
+        digits_compute=dp.get_precision('Product Unit of Measure'),
+    )
+
+    @api.constrains('res_user_id')
+    def _check_user(self):
+        for rec in self:
+            if self.env.user.has_group('stock.group_stock_user'):
```
Comment (rousseldenis) [inline stock_reminder/models/stock_reminder.py:44]: Put this before the loop.

File: stock_reminder/models/stock_reminder.py [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+    def write(self, vals):
+        """ Make the user a follower of the product
+        """
+        res = super(StockReminder, self).write(vals)
+        model_product_product = self.env['product.product']
+        user_id = vals.get('res_user_id')
+        product_id = model_product_product.browse(
+            vals.get('product_product_id'))
+        if user_id and product_id:
+            model_product_product.browse(product_id).message_subscribe_users(
```
Comment (rousseldenis) [inline stock_reminder/models/stock_reminder.py:72]: Why re-browsing product ?

File: stock_reminder/models/stock_reminder.py [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+    @api.model
+    def _check_reminders(self):
+        """
+        Go through all the reminders and see if they evaluate to true.
+        If yes go ahead and send reminders to the interested users.
+        """
+        for rec in self.search([]):
+            qty_available = rec.product_product_id._product_available()[
+                rec.product_product_id.id]['qty_available']
+            result = safe_eval(
```
Comment (rousseldenis) [inline stock_reminder/models/stock_reminder.py:91]: You cannot do that in that way. You need to use float_compare method.

File: stock_reminder/models/stock_reminder.py [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+                        'You can only set reminders for yourself.'))
+
+    @api.model
+    def create(self, vals):
+        """ Make the user a follower of the product
+        """
+        res = super(StockReminder, self).create(vals)
+        partner_id = self.env['res.users'].browse(
+            vals.get('res_user_id')).partner_id
+        self.env['product.product'].browse(
```
Comment (rousseldenis) [inline stock_reminder/models/stock_reminder.py:56]: Why not using 'message_subscribe' as here under ?

**Review by rousseldenis — CHANGES_REQUESTED** [review]
Code review. Some required changes.

The worse point I see here, if there are some products that trigger often the reminder, the chatter on the product form could be bulky to manage.

**Comment by rousseldenis** [comment]
@daramousk Do you plan to attend comments ?

**Comment by daramousk** [comment]
@rousseldenis i will, if not today tomorrow

**Comment by rousseldenis** [comment]
@daramousk We are almost tomorrow :sweat_smile:

**Pull request closed without merging**